packages/eip712/src/encoding/utils.ts

```diff
@@ -0,0 +1,11 @@
+// Parse Chain ID string into number. Throws error on failure case.
+export function parseChainId(chainId: string): number {
+  const matches = chainId.match('_(.*)-')
```

Comment:
there's a regex on Ethermint that you can use

Reply:
Can you send a link? I'll look into it.

Reply:
```suggestion
  const matches = chainId.match('^([a-z]{1,})_{1}([1-9][0-9]*)-{1}([1-9][0-9]*)$')
```